driver-sync/src/main/com/mongodb/client/internal/MongoClientImpl.java

```diff
@@ -65,14 +68,24 @@ public MongoClientImpl(final Cluster cluster, final MongoClientSettings settings
                            @Nullable final OperationExecutor operationExecutor) {
         this.settings = notNull("settings", settings);
         AutoEncryptionSettings autoEncryptionSettings = settings.getAutoEncryptionSettings();
-        this.delegate = new MongoClientDelegate(notNull("cluster", cluster),
+        CodecRegistry codecRegistry = settings.getCodecRegistry();
+        if (settings.getUuidRepresentation() != UuidRepresentation.JAVA_LEGACY) {
+            if (settings.getCodecRegistry() instanceof CodecProvider) {
+                codecRegistry = new UuidRepresentationOverridingCodecRegistry((CodecProvider) settings.getCodecRegistry(),
+                        settings.getUuidRepresentation());
+            } else {
+                throw new MongoClientException("Changing the default UuidRepresentation requires a CodecRegistry that also implements the" +
```

Comment:
This is the main wart in the design.  See the PR description for the rationale

Reply:
This is a wart and stopping the world initially seemed harsh but the fix is clear enough from the exception.